Graphics/Implicit/Definitions.hs

```diff
@@ -229,104 +236,101 @@ data SharedObj obj vec
 
 instance (Show obj, Show vec) => Show (SharedObj obj vec) where
   showsPrec = flip $ \case
-     Empty                   -> showCon "emptySpace"
-     Full                    -> showCon "fullSpace"
-     Complement obj          -> showCon "complement"   @| obj
-     UnionR 0 l_obj          -> showCon "union"        @| l_obj
-     UnionR r l_obj          -> showCon "unionR"       @| r   @| l_obj
-     DifferenceR 0 obj l_obj -> showCon "difference"   @| obj @| l_obj
-     DifferenceR r obj l_obj -> showCon "differenceR"  @| r   @| obj @| l_obj
-     IntersectR 0 l_obj      -> showCon "intersect"    @| l_obj
-     IntersectR r l_obj      -> showCon "intersectR"   @| r   @| l_obj
-     Translate vec obj       -> showCon "translate"    @| vec @| obj
-     Scale vec obj           -> showCon "scale"        @| vec @| obj
-     Mirror vec obj          -> showCon "mirror"       @| vec @| obj
-     Outset r obj            -> showCon "outset"       @| r   @| obj
-     Shell r obj             -> showCon "shell"        @| r   @| obj
-     EmbedBoxedObj _         -> showCon "implicit"     @| Blackhole
-     WithRounding r obj      -> showCon "withRounding" @| r   @| obj
-
+    Empty -> showCon "emptySpace"
+    Full -> showCon "fullSpace"
+    Complement obj -> showCon "complement" @| obj
+    UnionR 0 l_obj -> showCon "union" @| l_obj
+    UnionR r l_obj -> showCon "unionR" @| r @| l_obj
+    DifferenceR 0 obj l_obj -> showCon "difference" @| obj @| l_obj
+    DifferenceR r obj l_obj -> showCon "differenceR" @| r @| obj @| l_obj
+    IntersectR 0 l_obj -> showCon "intersect" @| l_obj
+    IntersectR r l_obj -> showCon "intersectR" @| r @| l_obj
+    Translate vec obj -> showCon "translate" @| vec @| obj
+    Scale vec obj -> showCon "scale" @| vec @| obj
+    Mirror vec obj -> showCon "mirror" @| vec @| obj
+    Outset r obj -> showCon "outset" @| r @| obj
+    Shell r obj -> showCon "shell" @| r @| obj
+    EmbedBoxedObj _ -> showCon "implicit" @| Blackhole
+    WithRounding r obj -> showCon "withRounding" @| r @| obj
 
 ------------------------------------------------------------------------------
+
 -- | A type whose show instance is a hole @_@. Used for giving 'Show' instances
 -- to data types which contain functions or other unshowable things.
 data Blackhole = Blackhole
 
 instance Show Blackhole where
   show _ = "_"
 
-
 newtype ObjectContext = ObjectContext
   { objectRounding :: ℝ
-  } deriving (Eq, Ord, Show)
-
-defaultObjectContext :: ObjectContext
-defaultObjectContext = ObjectContext
-  { objectRounding = 0
   }
+  deriving (Eq, Ord, Show)
 
+defaultObjectContext :: ObjectContext
+defaultObjectContext =
+  ObjectContext
+    { objectRounding = 0
+    }
 
 -- | A symbolic 2D object format.
 --   We want to have symbolic objects so that we can
 --   accelerate rendering & give ideal meshes for simple
 --   cases.
-data SymbolicObj2 =
```

Comment:
this also is less clear.

Reply:
Comments are hard, since the parse tree contains incomplete information about which nodes a comment belongs to.

This requires some change in behavior to get right: you save, realize that your comments has moved to the wrong place, rewrite things a little bit, and after one or two tries the comment will stick where it belongs.